### ISSUE TITLE
环境变量的总大小超过 4KB 导致无法部署

### ISSUE THREAD

**Issue by hmsjy2017** [spencerwooo/onedrive-vercel-index#19]
![AC177B45-35E0-4788-8BC1-7D3954BAC27F](https://user-images.githubusercontent.com/42692274/125184717-b06c1100-e252-11eb-900f-97e786cc574d.jpeg)

**Comment by spencerwooo** [comment]
看来还是得用多个环境变量手动拼接的办法解决了，理论上和之前项目的解决方案一致。

**Comment by hmsjy2017** [comment]
@spencerwooo 就算手动拼接，总大小还是超过 4KB，可能得想别的办法了

**Comment by hmsjy2017** [comment]
@spencerwooo 把仓库设为私有，然后把access_token直接写在pages/api/index.ts里可以临时解决。

**Comment by anymeofu** [comment]
Same Issue for me. What's the Workaround you used ?

**Comment by spencerwooo** [comment]
@anymeofu he hard coded the secret into the source code directly and set the repo to private, tho i personally don't recommend this workaround.

**Comment by spencerwooo** [comment]
> @spencerwooo 就算手动拼接，总大小还是超过 4KB，可能得想别的办法了

我没注意到 Vercel 是有总环境变量大小的限制……这确实麻烦了

**Comment by anymeofu** [comment]
> @anymeofu he hard coded the secret into the source code directly and set the repo to private, tho i personally don't recommend this workaround.

So Vercel's Limitations cant be fixed, every time I generate accesstoken and refresh token, its more than 4kb in total..
Gonna stick to cloudflare one for a while...

**Comment by hmsjy2017** [comment]
@spencerwooo 可以试试用 Netlify 或者函数计算部署

**Comment by nfway** [comment]
给后来者一个参考，部署到 netlify 需要的修改：

修改 `next.config.js` 在倒数第二个 `}` 加上

target: 'experimental-serverless-trace'

在部署时手动加上 自定义变量 ：ACCESS_TOKEN、CLIENT_SECRET、REFRESH_TOKEN

**Comment by anymeofu** [comment]
> To give a reference to the latecomers, the modifications needed to deploy to netlify:
> 
> 1. Modify `next.config.js`the penultimate `}`plus
> 
> `target: 'experimental-serverless-trace' `
> 
> 1. Manually add custom variables during deployment: ACCESS_TOKEN, CLIENT_SECRET, REFRESH_TOKEN
> 2. Build command:, `next build`leave the others blank

Can You explain In English,? Translation is not helping...

**Comment by hmsjy2017** [comment]
@anymeofu Please modify `next.config.js` to the following:
```
module.exports = {
  webpack: (config) => {
    // load worker files as a urls with `file-loader`
    config.module.rules.unshift({
      test: /pdf\.worker\.(min\.)?js/,
      use: [
        {
          loader: "file-loader",
          options: {
            name: "[contenthash].[ext]",
            publicPath: "_next/static/worker",
            outputPath: "static/worker"
          }
        }
      ]
    });

    return config;
  },
  reactStrictMode: true,
  images: {
    domains: ['public.dm.files.1drv.com'],
  },
  target: 'experimental-serverless-trace'
}
```

**Comment by hmsjy2017** [comment]
@anymeofu Please use Netlify instead of Vercel.

**Comment by anymeofu** [comment]
> @anymeofu I recommend you to use Netlify instead of Vercel. Please add ACCESS_TOKEN, CLIENT_SECRET three environment variables when building it. The build command is `next build`.
> 
> If you really need to use Vercel, you can change lines 15 to 20 in pages/api/index.ts to:
> 
> ```
> let _access_token ='Please fiill in your ACCESS_TOKEN here'
> const getAccessToken = async () => {
> // if (_access_token) {
> // console.log('Fetch token from memory.')
> // return _access_token
> //}
> ```
> 
> In order to prevent the access token from leaking, make sure to make the repository private.

thanks worked, will try on netlify

**Comment by transflo** [comment]
关于如何在Vercel部署：1.先将此仓库导入到自己的仓库，并将自己的这个仓库设为私有。2.修改pages/api/index.ts如下，写死accesstoken 3. config/api.json填入clientid  4.部署到vercel的时候只填refreshtoken和clientsecret
```
let _access_token ='Please fiill in your ACCESS_TOKEN here'
const getAccessToken = async () => {
// if (_access_token) {
// console.log('Fetch token from memory.')
// return _access_token
//}
```

**Comment by qkqpttgf** [comment]
@spencerwooo  可以部署我的OneManager到Vercel感受下操作过程。

根据 https://vercel.com/docs/api#endpoints/deployments/create-a-new-deployment ，可以拿token部署代码，也可以改变环境变量，但改变环境变量后，还是要重新部署一次才能生效，所以现在我干脆直接把代码跟config写/tmp中，直接部署。
这样issue22也解决了。

**Comment by spencerwooo** [comment]
Fixed by PR #210
